src/FixedDataTableRow.js

```diff
@@ -295,11 +301,18 @@ class FixedDataTableRowImpl extends React.Component {
   };
 
   _onMouseEnter = (/*object*/ event) => {
+    /**
+     * This is necessary so that onMouseLeave is fired with the initial
+     * row index since this row could be updated with a different index
+     * when scrolling.
+     */
+    this.mouseLeaveIndex = this.props.index;
     this.props.onMouseEnter(event, this.props.index);
   };
 
   _onMouseLeave = (/*object*/ event) => {
-    this.props.onMouseLeave(event, this.props.index);
+    this.props.onMouseLeave(event, this.mouseLeaveIndex);
```

Comment:
Is there any risk this.mouseLeaveIndex could be null here?  And in that case do we want to fall back to this.props.index?

I haven't played around w/ this enough to know if that's possible or not.  I'll try some today though.

Reply:
On a second thought I think there might be that risk if someone defines a function for onMouseLeave but not onMouseEnter. In that case this.mouseLeaveIndex would never be assigned and always be null. Changing to this.mouseLeaveIndex || this.props.index.

Reply:
Defining onMouseLeave but not onMouseEnter creates another weird issue existing in the existing codebase as well:
- Mouse hover on a row
- Without moving the mouse scroll over to another row
- Move the mouse so that you enter the new row
- onMouseLeave is fired with a "random" index
- Move the mouse out of that row
- onMouseLeave is fired with the correct index

Not sure what the first mouse leave refers to.

Reply:
this.mouseLeaveIndex || this.props.index will probably not do what you want when this.mouseLeaveIndex is 0.  Better to explicitly check for null.

I'll look into the other weirdness

Reply:
Good point, didn't think about that. Changed to check specifically for null.

Reply:
Things look good for the case where both an onMouseEnter & onMouseLeave are declared.

For the weirdness, I think its an issue from the row being recycled & index changing.  We could go back to your original solution to solve it, but a better case may be to update the code so we bind onMouseEnter whenever onMouseLeave exists
```
onMouseEnter={this.props.onMouseEnter || this.props.onMouseLeave ? this._onMouseEnter : null}
```
and then we only call the onMouseEnter prop is it exists.

```
_onMouseEnter = (/*object*/ event) => {
    /**
     * This is necessary so that onMouseLeave is fired with the initial
     * row index since this row could be updated with a different index
     * when scrolling.
     */
    this.mouseLeaveIndex = this.props.index;
    if (this.props.onMouseEnter) {
      this.props.onMouseEnter(event, this.props.index);
    }
  };
```

Also do we need to handle any of these cases for onTouchStart & onTouchEnd?

Reply:
That sounds reasonable. Did you give it a try to see if the weirdness goes away with this approach?
Also, I don't think we need to worry about those touch events since the touchEnd is fired as soon as you release the finger from the screen.

Reply:
I ran this locally and it seemed to work (based on console logs)

Reply:
I gave it another go and it looks good to me. I committed the changes, for me it is good to go.